var/spack/repos/builtin/packages/llvm/package.py

```diff
@@ -324,6 +324,9 @@ class Llvm(CMakePackage):
     conflicts('+clang_extra', when='~clang')
     conflicts('+lldb',        when='~clang')
 
+    # Github issue #4986
+    patch('llvm_gcc7.patch', when='@4.0.0:4.0.1+lldb ^gcc@7.0:')
```

Comment:
I may be missing something, but:
```console
$ spack spec -Il  llvm@4.0.0 ^gcc@7.1.0
Input spec
--------------------------------
     gkeax7s  llvm@4.0.0
     zw2hms6      ^gcc@7.1.0

Normalized
--------------------------------
==> Error: llvm does not depend on gcc
```
Maybe you wanted to say `when='@4.0.0:4.0.1+lldb %gcc@7.0:'? (Notice the `%` instead of the `^`)

Reply:
@alalazo Sorry, you are right :) I just wanted to show that it can be installed with gcc and forgot how :) I actually had `gcc@7.1.0` specified in my `packages.yaml`

Reply:
Oh, and excellent catch @alalazo. This sounds like a real issue to me.